### PR TITLE
otelhttp: add WithClientTrace to connect with otelhttptrace 

### PR DESCRIPTION
**Pull request by tonistiigi**
I couldn't quite figure out how `otelhttp` and `otelhttptrace` are supposed to be used together. From they being in separate packages I assume maintainers don't want them to depend on each other.

This adds a new option so caller can initialize `Transport` with `otelhttp.NewTransport(base, otelhttp.WithClientTrace(otelhttptrace.NewClientTrace))` (or any other `httptrace` implementation) so that `otelhttp` spans will have `otelhttptrace` spans as children.

Depends on #873

### CODE REVIEW

**Comment by thaJeztah** [comment]
@tonistiigi https://github.com/open-telemetry/opentelemetry-go-contrib/pull/873 was merged; guess this needs to be rebased and moved out of draft

**Comment by thaJeztah** [comment]
https://github.com/open-telemetry/opentelemetry-go-contrib/pull/874 was merged as well; this ready to be moved out of draft?

**Comment by thaJeztah** [comment]
@pellared  @MadVikingGod @Aneurysm9 ptal; this is the remaining one for us to get rid of our fork ❤️

File: instrumentation/net/http/otelhttp/transport.go
Comment (errordeveloper) [inline instrumentation/net/http/otelhttp/transport.go:110]: Just out of curiosity, would having a default client trace be helpful? would that be considered an API breakage at this point?

**Review by Aneurysm9 — APPROVED** [review]

File: instrumentation/net/http/otelhttp/transport.go
Comment (Aneurysm9) [reply instrumentation/net/http/otelhttp/transport.go:110]: I don't think it would be API breaking, but perhaps a surprising change in behavior.  I'm not sure a default client trace is a good idea, either, as it adds overhead and verbosity that is not necessary in many cases.

Comment (errordeveloper) [reply instrumentation/net/http/otelhttp/transport.go:110]: Makes sense, thanks. Very helpful, as always!

**Comment by thaJeztah** [comment]
Looks like it may need another rebase

**Comment by Aneurysm9** [comment]
This still requires another approval.  @open-telemetry/go-approvers can anyone else weigh in on this?

**Comment by evantorrie** [comment]
LGTM 👍

**Comment by thaJeztah** [comment]
Thank you!

**Pull request merged**
